### ISSUE TITLE
Read SDCs before `place_design` in non-leaf hierarchical flows

### ISSUE THREAD

**Issue by jerryz123** [ucb-bar/hammer#820]
### Background work

- [X] Yes, I searched the [documentation](https://hammer-vlsi.readthedocs.io/)
- [X] Yes, I searched in the [Support (Q&A)](https://github.com/ucb-bar/hammer/discussions/categories/support-q-a) section of the Discussions and read the [posting guidelines](https://github.com/ucb-bar/hammer/discussions/717)
- [X] Yes, I searched [prior issues](https://github.com/ucb-bar/hammer/issues?q=)
- [X] This issue is specific to Hammer, i.e., does not belong in a plugin repository instead
- [X] My report does not contain sensitive info (e.g., PDK information under NDA)
- [ ] (If applicable) Yes, I searched the [Chipyard documentation](https://chipyard.readthedocs.io/)
- [ ] (If applicable) Yes, I searched the [Chipyard mailing list](https://groups.google.com/forum/#!forum/chipyard)
- [ ] (If applicable) Yes, I searched the [Chipyard issues](https://github.com/ucb-bar/chipyard/issues?q=)

### Hammer version and plugin hashes

Release: 1.0.0
Hash: a1b2c3

### Other Setup

Ex: OS information, steps taken, documentation followed, etc.

### Current Behavior

Currently, the behavior for non-leaf clock constraints in PAR follows the workaround introduced in this old PR: https://github.com/ucb-bar/hammer-cadence-plugins/pull/63

To summarize:
 - In some cases, bottom-up hierarchical non-leaf PAR would fail to read the SDCs before CTS, causing errors.
 - The work around in https://github.com/ucb-bar/hammer-cadence-plugins/pull/63 was to force updating the constraint mode before CTS

### Expected Behavior

Two things need to be investigated:
 1) Is the original issue still present in recent Innovus versions? I believe this is the case, since we are seeing non-fatal error messages about hierarchical PAR failing to find clocks before CTS stage ... but it would be good to confirm
 2) Can `update_constraint_mode` be moved before `place_design`, so it can inform timing-aware placement?
